src/sorted_dict.jl

```diff
@@ -3,411 +3,294 @@
 
 """
-    findkey(sd, k)
+    findkey(sd::SortedDict, k)
 
-Argument `sd` is a SortedDict and argument `k` is a key. This
-function returns the semitoken that refers to the item whose key is
-`k`, or past-end semitoken if `k` is absent. Time: O(*c* log *n*)
+Return the semitoken that 
+points to the item whose key is
+`k`, or past-end semitoken if `k` is absent. 
+See also  [`Base.getindex(sd::SortedDict, k)`](@ref)
+Time: O(*c* log *n*)
 """
 @inline function findkey(m::SortedDict, k_)
     ll, exactfound = findkey(m.bt, convert(keytype(m),k_))
     IntSemiToken(exactfound ? ll : 2)
 end
 
-## This function inserts an item into the tree.
-## Unlike m[k]=d, it also returns a bool and a token.
-## The bool is true if the inserted item is new.
-## It is false if there was already an item
-## with that key.
-## The token points to the newly inserted item.
 
 """
-    insert!(sc, k)
+    sd_push!(sd::SortedDict, p::Pair)
 
-Argument `sc` is a SortedDict or SortedMultiDict, `k` is a key and
-`v` is the corresponding value. This inserts the `(k,v)` pair into
-the container. If the key is already present in a SortedDict, this
-overwrites the old value. In the case of SortedMultiDict, no
-overwriting takes place (since SortedMultiDict allows the same key
-to associate with multiple values). In the case of SortedDict, the
-return value is a pair whose first entry is boolean and indicates
+Insert pair `p` of the form `k=>v` into `sd`.
+If the key is already present in `sd`, this
+overwrites the old value. Note that the key is also overwritten,
+which is not necessarily a no-op because equivalence in the sort
+order does not necessarily imply equality.  Unlike `push!`,
+the
+return value is a 2-tuple whose first entry is boolean and indicates
 whether the insertion was new (i.e., the key was not previously
-present) and the second entry is the semitoken of the new entry. In
-the case of SortedMultiDict, a semitoken is returned (but no
-boolean). Time: O(*c* log *n*)
+present) and whose second entry is the semitoken of the new entry.
+This function replaces the deprecated `insert!(sd,k,v)`.
+ Time: O(*c* log *n*)
 """
-@inline function Base.insert!(m::SortedDict{K,D,Ord}, k_, d_) where {K,D, Ord <: Ordering}
-    b, i = insert!(m.bt, convert(K,k_), convert(D,d_), false)
+@inline function sd_push!(m::SortedDict{K,D,Ord}, pr::Pair) where {K,D, Ord <: Ordering}
+    b, i = insert!(m.bt, convert(K,pr.first), convert(D,pr.second), false)
     b, IntSemiToken(i)
 end
 
-"""
-    eltype(sc)
-
-Returns the (key,value) type (a 2-entry pair, i.e., `Pair{K,V}`) for
-SortedDict and SortedMultiDict. Returns the key type for SortedSet.
-This function may also be applied to the type itself. Time: O(1)
-"""
-@inline Base.eltype(m::SortedDict{K,D,Ord}) where {K,D,Ord <: Ordering} =  Pair{K,D}
-@inline Base.eltype(::Type{SortedDict{K,D,Ord}}) where {K,D,Ord <: Ordering} =  Pair{K,D}
-
-"""
-    in(p, sc)
+@deprecate insert!(m::SortedDict, k, d) sd_push!(m::SortedDict, k=>d)
```

Comment:
can we move this into `src/deprecations.jl` so we remember to delete it in next release?

Reply:
Done